core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceRepository.scala

```diff
@@ -16,9 +27,20 @@ import scala.concurrent.Future
   */
 class WorkspaceRepository(dataSource: SlickDataSource) {
 
-  def getWorkspace(workspaceId: UUID): Future[Option[Workspace]] =
+  def getWorkspace(workspaceId: UUID): Future[Option[Workspace]] = getWorkspace(workspaceId, None)
+
+  def getWorkspace(workspaceId: UUID, attributeSpecs: Option[WorkspaceAttributeSpecs]): Future[Option[Workspace]] =
     dataSource.inTransaction { access =>
-      access.workspaceQuery.findById(workspaceId.toString)
+      access.workspaceQuery.findV2WorkspaceById(workspaceId, attributeSpecs)
+    }
+
+  def getWorkspace(workspaceName: WorkspaceName): Future[Option[Workspace]] = getWorkspace(workspaceName, None)
+
+  def getWorkspace(workspaceName: WorkspaceName,
+                   attributeSpecs: Option[WorkspaceAttributeSpecs]
+  ): Future[Option[Workspace]] =
+    dataSource.inTransaction { dataAccess =>
+      dataAccess.workspaceQuery.findV2WorkspaceByName(workspaceName, attributeSpecs)
```

Comment:
I thought that transactions are used to combine multiple operations into a single transaction so, for example, they could be rolled back together. I am confused why a single query would need to be wrapped in a transaction.

Reply:
TBH, I never really considered it; I was just iterating on what was already there. I don't think I've seen any usage in the code base that wasn't wrapped in a transaction. 

Now it's kind of bugging me, and I want to go through and change all of the single call operations to `dataSource.dataAccess`.

Reply:
Perhaps it has to be wrapped in a transaction??? I just don't know.

Reply:
It doesn't _have_ to be: I checked, and you can get the `dataAccess` object outside of the transaction block.

That said, since there isn't anywhere else in the code base using it directly (at least that I'm aware of), I'd want to be a bit more intentional about trying it, and make sure I fully understood any implications first.